server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -421,6 +420,7 @@ private class EmailMetadataViewFactory @Inject()(zoneIdProvider: ZoneIdProvider)
     val mailboxIds: MailboxIds = MailboxIds(message._2
       .map(_.getMailboxId)
       .toList)
+    val threadId: ThreadId = ThreadId(message._2.head.getThreadId.getBaseMessageId.serialize())
```

Comment:
We should not expose at the JMAP level `getBaseMessageId` which is an implementation detail.

We should have a `ThreadId::serialize` method IMO